best_practices/security.rst

```diff
@@ -173,6 +182,25 @@ the same ``getAuthorEmail`` logic you used above:
 
             return false;
         }
+
+        /**
+         * Checks if the user token has the given role taking into account the
+         * entire role hierarchy if defined by the application.
+         */
+        private function userHasRole(UserInterface $user, $roleName)
+        {
+            if (null === $this->roleHierarchy) {
+                return in_array($roleName, $user->getRoles(), true);
```

Comment:
this is wrong. You should be using the roles from the token to be consistent with the way the RoleVoter is working